qcodes/dataset/plotting.py

```diff
@@ -45,22 +48,31 @@ def set_axis_labels(ax, data):
        * 1D plots
        * 2D plots on filled out rectangular grids
     """
-
     alldata = get_data_by_id(run_id)
-
-    for data in alldata:
+    nplots = len(alldata)
+    if isinstance(axes, mplaxes):
+        axes = [axes]
+
+    if axes is None:
+        axes = []
+        for i in range(nplots):
+            fig, ax = plt.subplots(1,1)
+            axes.append(ax)
+    else:
+        if len(axes) != nplots:
+            raise RuntimeError(f"Trying to plot {nplots} but "
```

Comment:
Consider the rewording `f"Trying to make {nplots} plots, but received {len(axes)} {ax_word}."`
Where `ax_word = 'axis' if len(axes) == 1 else 'axes' :)

Reply:
Agreed apart from the last part. In matplotlib  API an Axes object is the thing that holds both the x and y axis, ax axis is either the x or y axis. That is confusing but the way it works